src/aap_eda/core/utils/credentials.py

```diff
@@ -267,13 +273,33 @@ def validate_injectors(schema: dict, injectors: dict) -> dict:
 
         for k, v in input_data.items():
             try:
+                if k in key_names:
+                    raise InjectorDuplicateKey(
+                        f"Injector {field} key: {k} already exists"
+                    )
+
+                if field == "file":
+                    keys = k.split(".")
+                    if keys[0] != "template":
+                        raise InjectorInvalidTemplateKey(
+                            f"Injector {field} key: {k} "
+                            "should start with template"
+                        )
+                    if len(keys) > 2:
```

Comment:
@jshimkus-rh This is the server side validation of the keys to be either 
* template
* template.cafile
* template.certfile
This will get rejected
* template.cafile.text